### ISSUE TITLE
Enhancement: Improve guessing the URL to the latest refpages version

### ISSUE THREAD

**Issue by TSlivede** [KhronosGroup/OpenCL-Docs#308]
Followup on https://github.com/KhronosGroup/OpenCL-Docs/issues/251

@Khronoswebmaster 
Minor issue:

I guess you added some kind of link (named `latest`) to the www.khronos.org/registry/OpenCL/sdk/ folder, that redirects to the `2.2` folder.

In the same way, could you add a link named `xhtml` to the www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/ folder, that redirects to the `html` folder?

This way, if one does not have a "latest" link, one can simply change the version number to `latest`, and it works. If one currently changes the version number in an old link to `latest`, one gets redirected, to an inexistent location.

Example:

Some link led me to this webpage:
www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clGetDeviceIDs.html

Assuming I am a somewhat "technologically affine" person, I think "oh, maybe there is a newer version available, let's try changing the version number in the URL to `latest`" (yes, this strategy works quite often) - I then get the page:
www.khronos.org/registry/OpenCL/sdk/latest/docs/man/xhtml/clGetDeviceIDs.html

That link automatically redirects to the invalid location:
www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/xhtml/clGetDeviceIDs.html

If the `xhtml` folder would now automatically redirect to `html`, it would have worked.

-----

For the same reason (guessing the URL), could you redirect www.khronos.org/registry/OpenCL/sdk/docs/ to www.khronos.org/registry/OpenCL/sdk/latest/ ?
If one removes the version number from    
www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clGetDeviceIDs.html    
one gets    
www.khronos.org/registry/OpenCL/sdk/docs/man/xhtml/clGetDeviceIDs.html    
which currently also doesn't work...
(The strategy "remove the version number from the URL to get the latest version" also works for many pages...)

-----

tl;dr;

Please redirect
www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/xhtml/ to www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/ and    
www.khronos.org/registry/OpenCL/sdk/docs/ to www.khronos.org/registry/OpenCL/sdk/latest/ !

**Comment by Khronoswebmaster** [comment]
Currently we rewrite:   
OpenCL/sdk/latest/* -> OpenCL/sdk/2.2/$1

Perhaps better to rewrite:  
OpenCL/sdk/latest/docs/man/x?html/* -> OpenCL/sdk/2.2/docs/man/html/*

For the idea of removing the version entirely, can you send an example of a URL where this works?

**Comment by TSlivede** [comment]
>send an example of a URL where this works

Do you mean something like this?    
Old version number: https://docs.python.org/2.7/library/intro.html
Number removed:    https://docs.python.org/library/intro.html ->redirects to https://docs.python.org/3/library/intro.html

**Comment by outofcontrol** [comment]
Thanks. I mis-read your original request, I thought you were saying it sometimes worked on our site.

**Comment by TSlivede** [comment]
>OpenCL/sdk/latest/docs/man/x?html/* -> OpenCL/sdk/2.2/docs/man/html/*

That would prevent accessing the parent directories with a link like this: www.khronos.org/registry/OpenCL/sdk/latest/

The rewrite rule could instead be:
```
RewriteRule "OpenCL/sdk/(latest/(docs/)?|(docs/))(man/)?x?(html/)?(.*)" "OpenCL/sdk/2.2/$2$3$4$5$6"
```

If I got the syntax correctly, this should keep these links working the same way as they are working now:    
```
www.khronos.org/registry/OpenCL/sdk/latest/docs/man/html/     ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/
www.khronos.org/registry/OpenCL/sdk/latest/docs/man/          ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/
www.khronos.org/registry/OpenCL/sdk/latest/docs/              ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/
www.khronos.org/registry/OpenCL/sdk/latest/                   ->   www.khronos.org/registry/OpenCL/sdk/2.2/
www.khronos.org/registry/OpenCL/sdk/                          ->   www.khronos.org/registry/OpenCL/sdk/
```

And it should also make these links work in a sensible way:
```
www.khronos.org/registry/OpenCL/sdk/latest/docs/man/xhtml/    ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/
www.khronos.org/registry/OpenCL/sdk/docs/man/xhtml/           ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/
www.khronos.org/registry/OpenCL/sdk/docs/man/html/            ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/
www.khronos.org/registry/OpenCL/sdk/docs/man/                 ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/
www.khronos.org/registry/OpenCL/sdk/docs/                     ->   www.khronos.org/registry/OpenCL/sdk/2.2/docs/
```
